packages/local_auth/local_auth_windows/lib/local_auth_windows.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:local_auth_platform_interface/local_auth_platform_interface.dart';
+import 'package:local_auth_platform_interface/types/auth_messages.dart';
+import 'package:local_auth_platform_interface/types/auth_options.dart';
+import 'package:local_auth_platform_interface/types/biometric_type.dart';
+import 'package:local_auth_windows/types/auth_messages_windows.dart';
+
+export 'package:local_auth_platform_interface/types/auth_messages.dart';
+export 'package:local_auth_platform_interface/types/auth_options.dart';
+export 'package:local_auth_platform_interface/types/biometric_type.dart';
+export 'package:local_auth_windows/types/auth_messages_windows.dart';
+
+const MethodChannel _channel =
+    MethodChannel('plugins.flutter.io/local_auth_windows');
+
+/// The implementation of [LocalAuthPlatform] for Windows.
+class LocalAuthWindows extends LocalAuthPlatform {
+  /// Registers this class as the default instance of [LocalAuthPlatform].
+  static void registerWith() {
+    LocalAuthPlatform.instance = LocalAuthWindows();
+  }
+
+  @override
+  Future<bool> authenticate({
+    required String localizedReason,
+    required Iterable<AuthMessages> authMessages,
+    AuthenticationOptions options = const AuthenticationOptions(),
+  }) async {
+    assert(localizedReason.isNotEmpty);
+    final Map<String, Object> args = <String, Object>{
+      'localizedReason': localizedReason,
+      'useErrorDialogs': options.useErrorDialogs,
+      'stickyAuth': options.stickyAuth,
+      'sensitiveTransaction': options.sensitiveTransaction,
+      'biometricOnly': options.biometricOnly,
+    };
+    args.addAll(const WindowsAuthMessages().args);
+    for (final AuthMessages messages in authMessages) {
+      if (messages is WindowsAuthMessages) {
+        args.addAll(messages.args);
+      }
+    }
+    return (await _channel.invokeMethod<bool>('authenticate', args)) ?? false;
+  }
+
+  @override
+  Future<bool> deviceSupportsBiometrics() async {
+    return (await getEnrolledBiometrics()).isNotEmpty;
```

Comment:
This is broken; we fixed it recently in iOS and Android (where it was a regression from federation); it's supposed to call the method channel invocation directly so that it can pick up 'undefined'.

But the whole 'undefined' thing was awful in the first place, and we're moving away from it. Instead, you should define a new method channel call for `deviceSupportsBiometrics`, and call that here, and get rid of 'undefined' anywhere it's referenced.

Reply:
Fixed.